# 00 · This laptop, from scratch
Inspect Python, RAM, CPU, NVIDIA driver and PyTorch before building any data. The portable workflow uses CPU by default and does not need Java or Spark.

In [1]:
from pathlib import Path
import sys
ROOT = next(p for p in [Path.cwd(), *Path.cwd().parents] if (p / 'scripts/data_pipeline.py').exists())
if str(ROOT) not in sys.path:
    sys.path.insert(0, str(ROOT))
from IPython.display import display, Image
import pandas as pd
from scripts.data_pipeline import OUT, REPORTS


In [2]:
from scripts.project_analysis import hardware
hardware_report = hardware()
display(pd.Series(hardware_report, name="Current machine"))

{
  "python": "3.13.0 (tags/v3.13.0:60403a5, Oct  7 2024, 09:38:07) [MSC v.1941 64 bit (AMD64)]",
  "python_executable": "C:\\Users\\artix\\.venvs\\music\\Scripts\\python.exe",
  "platform": "Windows-11-10.0.26200-SP0",
  "cpu_logical_cores": 12,
  "ram_gib": 7.37,
  "available_ram_gib": 0.99,
  "free_disk_gib": 10.84,
  "torch_version": "2.13.0+cpu",
  "cuda_available": false,
  "torch_cuda_runtime": null,
  "default_device": "cpu",
  "worker_threads": 4,
  "nvidia_gpu": "GeForce GTX 1650, 4096 MiB, 457.49"
}


python                3.13.0 (tags/v3.13.0:60403a5, Oct  7 2024, 09:...
python_executable        C:\Users\artix\.venvs\music\Scripts\python.exe
platform                                      Windows-11-10.0.26200-SP0
cpu_logical_cores                                                    12
ram_gib                                                            7.37
available_ram_gib                                                  0.99
free_disk_gib                                                     10.84
torch_version                                                2.13.0+cpu
cuda_available                                                    False
torch_cuda_runtime                                                 None
default_device                                                      cpu
worker_threads                                                        4
nvidia_gpu                           GeForce GTX 1650, 4096 MiB, 457.49
Name: Current machine, dtype: object

The GTX 1650 has 4 GB VRAM. The original CUDA 13 configuration cannot run with driver 457.49. A CPU wheel works immediately; GPU use requires a compatible driver and PyTorch installation. Data stages run in separate kernels to release memory.